src/modules/user/services/user.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { FindOptionsWhere, Repository } from "typeorm";
+import { v4 as uuidv4 } from "uuid";
+import { customAlphabet } from "nanoid/async";
+import { User } from "../model/user.entity";
+import { UserNotFoundException } from "./exceptions";
+
+@Injectable()
+export class UserService {
+  constructor(@InjectRepository(User) private userRepository: Repository<User>) {}
+
+  public async createUserFromDiscordId({ discordId }: { discordId: string }) {
+    let user = await this.userRepository.findOne({ where: { discordId } });
+
+    if (!user) {
+      const shortId = await this.createShortId();
+      const uuid = this.createUUID();
```

Comment:
Why do we have 3 ids (the primary key, the short id and the uuid) that are all generated by our application? What are they used for?

Reply:
For now only the `id` is used, but I added the `uuid` and the `shortId` in case we need to identify and expose users by non-incremental and random identifiers.